src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java

```diff
@@ -140,59 +142,37 @@
-                        // add entry in exception table
-                        validationServiceFacade.logFailures(format(NO_USER_TO_SUSPEND, cwrRequest.getRowId()),
-                            cwrRequest.getRowId());
-                        continue;
-                    }
-                    //when profile is new then create new user profile
-                    caseWorkerProfile = createCaseWorkerProfile(cwrRequest);
-                    //Avoid select Insert for new Inserts
-                    setNewCaseWorkerProfileFlag(caseWorkerProfile);
-                    newCaseWorkerProfiles.add(caseWorkerProfile);
-
-                } else if (isTrue(caseWorkerProfile.getSuspended())) {
-                    //when existing profile with delete flag is true then log exception
-                    // add entry in exception table
-                    validationServiceFacade.logFailures(ALREADY_SUSPENDED_ERROR_MESSAGE, cwrRequest.getRowId());
-                } else if (cwrRequest.isSuspended()) {
-                    //when existing profile with delete flag is true in request then suspend user
-                    UserProfileUpdatedData usrProfileStatusUpdate = UserProfileUpdatedData.builder()
-                        .idamStatus(IDAM_STATUS_SUSPENDED).build();
-                    if (isUserSuspended(usrProfileStatusUpdate, caseWorkerProfile.getCaseWorkerId(),
-                        ORIGIN_EXUI, cwrRequest.getRowId())) {
-                        caseWorkerProfile.setSuspended(true);
-                        newCaseWorkerProfiles.add(caseWorkerProfile);
-                    }
-                } else {
-                    //when existing profile with delete flag is false then update user in CRD db and roles in SIDAM
-                    requestMap.put(caseWorkerProfile.getEmailId(), cwrRequest);
-                    updateCaseWorkerProfiles.add(caseWorkerProfile);
-                }
+            //remove all existing profiles requests from cwUiRequests to separate out new and update/suspend profiles
+            for (CaseWorkerProfile dbProfile : cwDbProfiles) {
+                cwUiRequests.remove(emailToRequestMap.get(dbProfile.getEmailId().toLowerCase()));
             }
 
-            newCaseWorkerProfiles = newCaseWorkerProfiles.stream()
-                .filter(Objects::nonNull).collect(Collectors.toList());
-            updateCaseWorkerProfiles = updateCaseWorkerProfiles.stream()
-                .filter(Objects::nonNull).collect(Collectors.toList());
-            processedCwProfiles = persistCaseWorkerInBatch(newCaseWorkerProfiles, updateCaseWorkerProfiles, requestMap);
+            //process new CW profiles
+            newCaseWorkerProfiles = processNewCaseWorkers(cwUiRequests);
+            //process update and suspend CW profiles
+            Pair<List<CaseWorkerProfile>, List<CaseWorkerProfile>> updateAndSuspendedLists = processExistingCaseWorkers(
+                    emailToRequestMap, cwDbProfiles);
```

Comment:
I think better to create our own Pair it confuses left & right. 

Pair {
List<CaseWorkerProfile> suspendedProfiles
List<CaseWorkerProfile> updatedProfiles

}

Reply:
I used the readymade given by apache , rather than creating new data structure :)